src/main/kotlin/com/makarytskyi/rentcar/service/OrderService.kt

```diff
@@ -3,84 +3,84 @@ package com.makarytskyi.rentcar.service
 import com.makarytskyi.rentcar.dto.order.CreateOrderRequest
 import com.makarytskyi.rentcar.dto.order.OrderResponse
 import com.makarytskyi.rentcar.dto.order.UpdateOrderRequest
-import com.makarytskyi.rentcar.exception.OrderNotFoundException
+import com.makarytskyi.rentcar.exception.ResourceNotFoundException
 import com.makarytskyi.rentcar.model.Order
-import com.makarytskyi.rentcar.model.User
 import com.makarytskyi.rentcar.repository.CarRepository
 import com.makarytskyi.rentcar.repository.OrderRepository
 import com.makarytskyi.rentcar.repository.UserRepository
 import java.util.Date
 import org.springframework.stereotype.Service
 
 @Service
-class OrderService(
+internal class OrderService(
     private val orderRepository: OrderRepository,
     private val carRepository: CarRepository,
     private val userRepository: UserRepository
 ) {
 
     fun getById(id: String): OrderResponse =
         orderRepository.findById(id)?.let { Order.toResponse(it, getCarPrice(it.carId)) }
-            ?: throw OrderNotFoundException("Order with id $id is not found")
+            ?: throw ResourceNotFoundException("Order with id $id is not found")
 
     fun findAll(): List<OrderResponse> =
         orderRepository.findAll().map { Order.toResponse(it, getCarPrice(it.carId)) }
             .toList()
 
     fun create(createOrderRequest: CreateOrderRequest): OrderResponse {
 
-        validateDatesOrder(createOrderRequest.from, createOrderRequest.to)
+        validateDates(createOrderRequest.from, createOrderRequest.to)
         validateUserExists(createOrderRequest.userId)
-        validateCarExists(createOrderRequest.carId)
         validateCarAvailability(createOrderRequest.carId, createOrderRequest.from, createOrderRequest.to)
 
-        return Order.toResponse(orderRepository.create(CreateOrderRequest.toEntity(createOrderRequest)), getCarPrice(createOrderRequest.carId))
+        return Order.toResponse(
+            orderRepository.create(CreateOrderRequest.toEntity(createOrderRequest)),
+            getCarPrice(createOrderRequest.carId)
+        )
     }
 
     fun deleteById(id: String) = orderRepository.deleteById(id)
 
-    fun findByUserId(userId: String): List<OrderResponse> = orderRepository.findByUserId(userId).map {
+    fun findByUser(userId: String): List<OrderResponse> = orderRepository.findByUserId(userId).map {
         Order.toResponse(it, getCarPrice(it.carId))
     }
 
-    fun findByCarId(carId: String): List<OrderResponse> = orderRepository.findByCarId(carId).map {
+    fun findByCar(carId: String): List<OrderResponse> = orderRepository.findByCarId(carId).map {
         Order.toResponse(it, getCarPrice(it.carId))
     }
 
-    fun update(id: String, orderRequest: UpdateOrderRequest): OrderResponse {
+    fun findByCarAndUser(carId: String, userId: String): List<OrderResponse> =
+        orderRepository.findByUserIdAndCarId(carId, userId)
+            .map { Order.toResponse(it, getCarPrice(it.carId)) }
```

Comment:
Domain should know nothing about response. Please invert. Call `OrderResponse.from(Order)`